### ISSUE TITLE
Nest build error when upgrading typescript to 4.8.X

### ISSUE THREAD

**Issue by leonard-henriquez** [nestjs/nest-cli#1812]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I have an error with the `nest build` command when I upgrade typescript from 4.7.4 to 4.8.4 (or any 4.8.X).
Sorry I didn't include a minimum reproduction code. I don't know I to build it because I have no details on where the error comes from.

<img width="601" alt="Capture d’écran 2022-10-21 à 13 47 57" src="https://user-images.githubusercontent.com/30215564/197189036-731dd343-6f8f-4151-bada-9ad919a225b4.png">

<img width="604" alt="Capture d’écran 2022-10-21 à 13 48 33" src="https://user-images.githubusercontent.com/30215564/197189078-363a8925-de35-40bc-826c-c549cce10cba.png">

I noticed the build works again if my remove all my CLI plugins (`@nestjs/swagger/plugin` and `@nestjs/graphql/plugin`).
It looks similar to https://github.com/nestjs/nest-cli/issues/1752 but all my Nest dependencies are at the last available versions.

### Minimum reproduction code

none

### Expected behavior

I would expect `nest build` to succeed

### Package version

9.1.4

### NestJS version

9.1.4

### Node.js version

16.17.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
Please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project).

**Comment by leonard-henriquez** [comment]
@kamilmysliwiec
As explained in my message, I would love to create a minimum reproduction repository but I haven't succeeded to do so because the error message is not clear enough to interpret where the error comes from.

I have tried both a top-down a a bottom-up approach to build such a example but I did not succeed:
- The top down approach from my repository: I tried to comment any part related to graphql but my codebase is too big so I havent succeeded to identify the part that created the bug
- The bottom up approach: I tried to build an example from the Nest js start on CodeSandbox. I added the dependencies (`@nestjs/graphql`) and tried to use it, but I could not reproduce the bug since I don't know which feature of `@nestjs/graphql` doesn't work

Is there a way I could have stack trace of the error to find which feature is responsible for this bug and create a minimum reproduction repository?

**Comment by kamilmysliwiec** [comment]
@leonard-henriquez 

1. Navigate to the "~/node_modules/@nestjs/swagger/dist/plugin/visitors/controller-class.visitor.js" file and add `console.log` as follows

![image](https://user-images.githubusercontent.com/23244943/197531469-7ef8f754-ae2f-4d20-9f27-f252e46742ce.png)

2. Navigate to the "~/node_modules/@nestjs/swagger/dist/plugin/visitors/model-class.visitor.js` and add console log there as welll

**Comment by kamilmysliwiec** [comment]
Are you sure that you added those console logs to the right package? Consider adding a console.log at the beginning of the file (either model-class or controller-class visitor) to see if it's even logged down when building your project. Make sure to also remove the dist folder before you rerun the build

**Comment by micalevisk** [comment]
@leonard-henriquez could you please disable only the `@nestjs/graphql` plugin?

**Comment by leonard-henriquez** [comment]
So I did try again adding a console.log at the beginning of the file.
The console.log is displayed but nothing more on the error @kamilmysliwiec ...

<img width="513" alt="Capture d’écran 2022-11-19 à 18 44 21" src="https://user-images.githubusercontent.com/30215564/202864335-738bc77d-4914-4de2-98fa-ff80d5cb328f.png">

Note:
- the error is still present with nest 9.2.0 and typescript 4.9.X
- I did disable graphql @micalevisk

**Comment by leonard-henriquez** [comment]
So, right after writing this message I manage to have a stack trace by adding a try catch in ControllerClassVisitor.visit.
The stack trace came from `updateImports` and I noticed it was using typescript 4.7.4 which was odd since my objective was to use typescript 4.9.4.
I went to my pnpm-lock.yaml and noticed the typescript version of Nest.js was not the same as the version of my monorepo.
When i upgraded the typescript version of my Nest.js app to 4.9.4 (the same as the rest of the monorepo) the error was fixed.
